[JSC] Update test262 to 01/29/2024

sosukesuzuki opened the pull request:
#### 7bee87603be3aa502b12d37d12ebaa7089df68b3
<pre>
[JSC] Update test262 to 01/29/2024
<a href="https://bugs.webkit.org/show_bug.cgi?id=286206">https://bugs.webkit.org/show_bug.cgi?id=286206</a>

Reviewed by NOBODY (OOPS!).

Update test262 to <a href="https://github.com/tc39/test262/commit/e7a84b0a090ca86cc05f2e90116115c8c2a7c059">https://github.com/tc39/test262/commit/e7a84b0a090ca86cc05f2e90116115c8c2a7c059</a>

* JSTests/test262/config.yaml:
* JSTests/test262/expectations.yaml:
* JSTests/test262/test262-Revision.txt:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e0c1671d8570e8877163081d8fb743a15f5a9e04

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/87517 "206 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/7030 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/41892 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/92379 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/38257 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/7412 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/15199 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/67600 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25340 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/90519 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/5651 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/79200 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/47970 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/5437 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/37372 "Built successfully") | 
| [❌ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/80312 "Found 768 new JSC stress test failures: microbenchmarks/abc-forward-loop-equal.js.ftl-eager-no-cjit, microbenchmarks/abs-boolean.js.ftl-eager-no-cjit, microbenchmarks/arguments-strict-mode.js.lockdown, microbenchmarks/arguments.js.ftl-eager-no-cjit, microbenchmarks/array-from-object-func.js.ftl-eager-no-cjit, microbenchmarks/array-from-object.js.ftl-eager-no-cjit, microbenchmarks/array-from-object.js.lockdown, microbenchmarks/array-prototype-concat-copy-double-and-int32.js.ftl-eager-no-cjit, microbenchmarks/array-prototype-concat-copy-obj.js.ftl-eager-no-cjit, microbenchmarks/array-prototype-sort-large-array-comparator-double.js.ftl-eager-no-cjit ... (failure)") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/75859 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/34458 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/94264 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/86291 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/14682 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/10779 "Found 1 new test failure: imported/w3c/web-platform-tests/html/capability-delegation/delegate-fullscreen-request-subframe-same-origin.https.tentative.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/76422 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/14936 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/75055 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/75649 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/20031 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/18455 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/7633 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/14700 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/108784 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/14443 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/26168 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/17887 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/16226 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Constellation reviewed (changes requested):
Did you run this on macOS Sequoia? I think Intl.DateTimeFormat etc. is not changed and removing them is incorrect.

MenloDorian reviewed (approved):
rs=me

sosukesuzuki committed:
[JSC] Update test262 to 01/29/2024
https://bugs.webkit.org/show_bug.cgi?id=286206

Reviewed by NOBODY (OOPS!).

Update test262 to https://github.com/tc39/test262/commit/e7a84b0a090ca86cc05f2e90116115c8c2a7c059

* JSTests/test262/config.yaml:
* JSTests/test262/expectations.yaml:
* JSTests/test262/test262-Revision.txt:
